### PR TITLE
gh-85525: Indicate supported sound header formats

### PR DESCRIPTION
**Pull request by nanjekyejoannah**
https://github.com/python/cpython/issues/85525

https://bugs.python.org/issue41353

* Issue: gh-85525

### CODE REVIEW

File: Doc/library/sndhdr.rst [hunk 1]
```diff
@@ -49,3 +49,51 @@ be the sample size in bits or ``'A'`` for A-LAW or ``'U'`` for u-LAW.
...
+| ``'8svx'`` | 8SVX Files                         |
++------------+------------------------------------+
+| ``'sb'``   | SB Files                           |
++------------+------------------------------------+
+| ``'ub'``   | UB Files                           |
++------------+------------------------------------+
+| ``'ul'``   | uLAW Audio Files                   |
++------------+------------------------------------+
+
+.. data:: tests
```
Comment (merwok) [inline Doc/library/sndhdr.rst:87]: Is this meant to be a public attribute?

File: Doc/library/sndhdr.rst [hunk 2]
```diff
@@ -49,3 +49,51 @@ be the sample size in bits or ``'A'`` for A-LAW or ``'U'`` for u-LAW.
...
 
+The following sound header types are recognized, as listed below with the return value
+from :func:`whathdr`: and :func:`what`:
+
++------------+------------------------------------+
+| Value      | Sound header format                |
++============+====================================+
+| ``'aifc'`` | Compressed Audio Interchange Files |
++------------+------------------------------------+
+| ``'aiff'`` | Audio Interchange Files            |
```
Comment (merwok) [inline Doc/library/sndhdr.rst:65]: Nice table!  Details: unnecessary capitalization for `files`; I think the table could be right at the beginning of the doc around line 31, and then you could remove the list of all values on lines 24-25.

File: Doc/library/sndhdr.rst (hunk 1 above)
Comment (nanjekyejoannah) [reply Doc/library/sndhdr.rst:87]: Looks like, please see imghdr.rst.

**Comment by hugovk** [comment]
Note the sndhdr module is deprecated in 3.11 and set for removal in 3.13.

See [PEP 594 – Removing dead batteries from the standard library](https://peps.python.org/pep-0594/) and https://github.com/python/cpython/issues/91217.

**Comment by AlexWaygood** [comment]
> Note the sndhdr module is deprecated in 3.11 and set for removal in 3.13.
> 
> See [PEP 594 – Removing dead batteries from the standard library](https://peps.python.org/pep-0594/) and #91217.

Nonetheless, this does look like a relatively simple docs enhancement that would be quite nice to have. It would be nice if it could get over the line.

File: Doc/library/sndhdr.rst (hunk 2 above)
Comment (nanjekyejoannah) [reply Doc/library/sndhdr.rst:65]: Am not a native english speaker but my impression is these are abbreviations hence capitalization, no? see, e.g https://en.wikipedia.org/wiki/Audio_Interchange_File_Format.

**Comment by nanjekyejoannah** [comment]
@willingc can help review it and see if it can be merged?

File: Doc/library/sndhdr.rst (hunk 2 above)
Comment (hugovk) [reply Doc/library/sndhdr.rst:65]: Yes, as the title of a format I think this is fine to capitalise. But let's use the same names as the formats:

* Compressed Audio Interchange File Format
* Audio Interchange File Format

And then consistently for the others, if possible. This was the best I found in some quick searches and may need double checking, some of these are really old!

* Au
* HCOM
* Sounder
* SndTool
* Creative Voice
* Waveform Audio File Format
* 8-Bit Sampled Voice
* Signed Byte Audio Data
* Unsigned Byte Audio File
* uLAW Audio File

I'm not too sure about `sb`, `ub` and `ul`: but more importantly looks like they were removed in 1999 and Python 2.0! 

https://github.com/python/cpython/commit/f4bf044999ad48684b8d70c229756a1da8a7613b

So let's ditch those and belatedly remove them from `SndHeaders.filetype.__doc__` and anywhere else :)

File: Doc/library/sndhdr.rst [hunk 3]
```diff
@@ -53,3 +53,51 @@ be the sample size in bits or ``'A'`` for A-LAW or ``'U'`` for u-LAW.
...
+   The test function should return a string describing the image type if the test
+   succeeded, or ``None`` if it failed.
+
+Example::
+
+   >>> import sndhdr
+   >>> imghdr.what('bass.wav')
+   'wav'
+   >>> imghdr.whathdr('bass.wav')
+   'wav'
```
Comment (hugovk) [inline Doc/library/sndhdr.rst]: When using "Python console" code blocks, we can use a special `pycon` to better highlight the syntax:


```suggestion
Example:

.. code-block:: pycon

   >>> import sndhdr
   >>> imghdr.what('bass.wav')
   'wav'
   >>> imghdr.whathdr('bass.wav')
   'wav'
```

**Comment by AA-Turner** [comment]
I agree with @AlexWaygood that this would be useful to consumers of the module for 3.10 through 3.12 -- I've no comments beyond Hugo's points though.

A

File: Doc/library/sndhdr.rst [hunk 4]
```diff
@@ -54,3 +54,52 @@ be the sample size in bits or ``'A'`` for A-LAW or ``'U'`` for u-LAW.
...
++============+====================================+
+| ``'aifc'`` | Compressed Audio Interchange Files |
++------------+------------------------------------+
+| ``'aiff'`` | Audio Interchange Files            |
++------------+------------------------------------+
+| ``'au'``   | Au Files                           |
++------------+------------------------------------+
+| ``'hcom'`` | HCOM Files                         |
++------------+------------------------------------+
++------------+------------------------------------+
```
Comment (merwok) [inline Doc/library/sndhdr.rst:71]: Why the empty row in the table?

Comment (nanjekyejoannah) [reply Doc/library/sndhdr.rst:71]: I just checked main, its not there, maybe another fix I dont know fixed it.

Comment (merwok) [reply Doc/library/sndhdr.rst:71]: How so?  I see it here https://github.com/python/cpython/blob/main/Doc/library/sndhdr.rst (and in raw view)

Comment (nanjekyejoannah) [reply Doc/library/sndhdr.rst:71]: I think I checked out the wring code version locally, my bad. I have a new PR to fix that : https://github.com/python/cpython/pull/98337

**Pull request merged**